packages/ui-dialog/src/Dialog/index.tsx

```diff
@@ -99,20 +96,11 @@ class Dialog extends Component<DialogProps> {
 
     this._raf.push(
       requestAnimationFrame(() => {
-        // It needs to wait a heartbeat until the content is fully loaded
-        // inside the dialog. If it contains a focusable element, it will
-        // get focused on open, and browsers scroll to the focused element.
-        // If the css is not fully applied, the element may not be in their
-        // final position, making the page jump.
-        this._timeouts.push(
-          setTimeout(() => {
-            this._focusRegion = FocusRegionManager.activateRegion(
-              this.contentElement as Element,
-              {
-                ...options
-              }
-            )
-          }, 0)
```

Comment:
Since react 18 rendering can be interrupted so a simple `setTimeout` doesn't guarantee that DOM will be completely up to date.